### PR TITLE
Simplified Nextcloud logo icon #2

### PR DESCRIPTION
**Pull request by pixelipo**
### Replaces pull request #5288

This is part of the SVG optimization task #5157 

Nextcloud logo code has been simplified, but at the same time it's more human-readable.

I've changed the size to more traditional values (256*128). Circles are a bit fatter than they used to be (+0.5px at 128px width). If that is an issue, I can change their width to 21.5px.

logo-icon.svg is deprecated since new logo.svg is already much smaller in weight.

I've combined .logo and .logo-icon - this is not strictly needed, I can revert if @nextcloud/designers disagree. It requires adding logo class to tags that have logo-icon class in apps - mainly where pages are shared publicly, e.g. Calendar.

PNG images and favicons were not edited.

Signed-off-by: Marin Treselj marin@pixelipo.com

### CODE REVIEW

File: core/css/header.scss [hunk 1]
```diff
@@ -100,29 +100,25 @@
...
 #header {
 	.logo {
-		background-image: url('#{$image-logo}');
+		display: inline-flex;
+		background-image: url($image-logo);
 		background-repeat: no-repeat;
-		background-size: 175px;
+		background-size: contain;
 		background-position: center;
-		width: 252px;
```
Comment (pixelipo) [inline core/css/header.scss]: As mentioned in #5288 if we want to limit the width of the logo to 175px, why not just make the container smaller? No need to check mime-type [here](files#diff-9930821d748dd9570b18f2ba7e400e3dL48)

Comment (juliushaertl) [inline core/css/header.scss]: The problem is that themed logos should use all of the space that is available, while the default logo should just have its default size.

Comment (pixelipo) [inline core/css/header.scss]: Ok, I see your point. I can make adjustments.
But in that case, can I ask why is the width set to 252px? parent div, `.header`, is 300px wide (probably in order to fit into smallest supported mobile width), while the form elements (`input`s and `submit`) are 269px wide. Seems arbitrary...

Comment (juliushaertl) [inline core/css/header.scss]: Maybe @jancborchardt has an idea about that?

**Comment by jancborchardt** [comment]
Yes, it is completely arbitrary and has some historical reasons I don't remember. The basic points are of course that:

- It should work well on mobile
- All the elements (input field and button) on log in should have the same width.
- Error messages need to be checked

If you can make that work, we can fix it all to 270 px or even 300px if that looks good.

**Comment by jancborchardt** [comment]
But maybe best let's merge this first, and do thaf in a separate pull request? Your call @pixelipo :)

**Comment by jancborchardt** [comment]
@pixelipo yes, that part should be reverted ideally because otherwise master is broken. :)

**Comment by pixelipo** [comment]
Should be ready to review/merge, @jancborchardt @juliushaertl @MorrisJobke

**Review by juliusknorr — APPROVED** [review]
I've rebased this to master. Works as expected. :+1:

**Comment by jancborchardt** [comment]
Lools good to me now - @juliushaertl @MorrisJobke?

**Comment by juliushaertl** [comment]
@jancborchardt Yes, already approved by me :wink: CI failing is unrelated.

**Review by MorrisJobke — APPROVED** [review]

**Pull request merged**